### PR TITLE
Update intersphinx mapping and sphinx-gallery-conf

### LINKED ISSUE #2471: Link in sphinx-gallery tutorials link to traitlets base class, not to ctapipe implementation
**Describe the bug**

If you go to e.g.:
https://ctapipe.readthedocs.io/en/latest/auto_examples/tutorials/calibrated_data_exploration.html
you see that sphinx-gallery creates cross-links from the code examples to documentation for classes.

Some classes. Only configuragbles. And the links don't point to the implementation of the specific class in ctapipe but to the 
`traitlets.config.Configurable` base class in the traitlets docs.

**To Reproduce**

Go to the above link and check.

**Expected behavior**

Link all ctapipe imports and link to the correct class, not the external base class.

### PR DESCRIPTION
**Pull request by Hckjs**
Links in tutorials should be fixed now. Fixes #2471, #2507

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -186,6 +186,9 @@ def setup(app):
     "default_thumb_file": "_static/ctapipe_logo.png",
     "pypandoc": True,
     "matplotlib_animations": True,
+    "reference_url": {
```
Comment (maxnoe) [inline docs/conf.py:189]: maybe add a comment: `# marks ctapipe as the current module for resolving intersphinx references`?

**Comment by maxnoe** [comment]
mmh, for some reason the readthedocs build didn't trigger in the last push...

**Review by maxnoe — APPROVED** [review]

File: docs/conf.py [hunk 2]
```diff
@@ -387,17 +391,25 @@ def setup(app):
...
-    "astropy": ("https://docs.astropy.org/en/latest/", None),
-    "pytables": ("https://www.pytables.org/", None),
     "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
-    "matplotlib": ("https://matplotlib.org/stable", None),
-    "cython": ("https://docs.cython.org/en/latest/", None),
-    "iminuit": ("https://scikit-hep.org/iminuit/", None),
+    "psutil": ("https://psutil.readthedocs.io/en/stable/", None),
+    "pytables": ("https://www.pytables.org/", None),
+    "pytest": ("https://docs.pytest.org/en/stable/", None),
+    "python": ("https://docs.python.org/3.12", None),
```
Comment (maxnoe) [inline docs/conf.py:409]: There is `https://docs.python.org/3/` which points to the latest 3

**Comment by Hckjs** [comment]
Thanks for the help & comments!

**Review by maxnoe — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**